### PR TITLE
feat(dapi): added auth lambda for tenovi

### PR DESCRIPTION
**Pull request by RamilGaripov**
refs. metriport/metriport#812

### Dependencies

- Upstream: _[993]_(https://github.com/metriport/metriport-internal/pull/993)

### Description

- Added an authorization lambda for Tenovi 

### Release Plan

- Merge the upstream dependency PR
- Add secrets manually
- Merge this and test on staging

### CODE REVIEW

File: packages/api/src/mappings/tenovi/body.ts [hunk 1]
```diff
@@ -6,31 +6,39 @@ import { getFloatValue, formatNumber } from "../../shared/numbers";
...
-  const weightSamples: Sample[] = [];
-  tenoviWeight.forEach(reading => {
-    weightSamples.push({
-      time: reading.timestamp,
-      value: formatNumber(convert(getFloatValue(reading.value_1)).from("lb").to("kg")),
+    const weightSamples: Sample[] = [];
+    tenoviWeight.forEach(reading => {
+      weightSamples.push({
+        time: reading.timestamp,
+        value: formatNumber(convert(getFloatValue(reading.value_1)).from("lb").to("kg")),
```
Comment (leite08) [inline packages/api/src/mappings/tenovi/body.ts:36]: NABD, but to help reading/maintaining this we could create some local constants with intermediate steps to get the final `value`

File: packages/infra/lib/api-stack.ts [hunk 2]
```diff
@@ -486,6 +486,18 @@ export class APIStack extends Stack {
       alarmAction: slackNotification?.alarmAction,
     });
 
+    this.setupTenoviWebhook({
```
Comment (leite08) [inline packages/infra/lib/api-stack.ts:489]: nit: Lambda instead of Webhook?

File: packages/infra/lib/api-stack.ts [hunk 3]
```diff
@@ -767,6 +779,58 @@ export class APIStack extends Stack {
...
+        API_URL: `http://${server.loadBalancer.loadBalancerDnsName}/webhook/tenovi`,
+        ENV_TYPE: envType,
+        TENOVI_AUTH_HEADER: tenoviAuthHeader,
+        ...(sentryDsn ? { SENTRY_DSN: sentryDsn } : {}),
+      },
+      vpc,
+      alarmSnsAction: alarmAction,
+    });
+
+    type SecretKeys = keyof EnvConfig["providerSecretNames"];
```
Comment (leite08) [inline packages/infra/lib/api-stack.ts:821]: Why this? I see its being used on `setupFitbitWebhook`, but it doesn't seem necessary - I just removed those local and it worked fine.

If we really don't need, let's remove it here and there while we're at it, please?

File: packages/lambdas/src/tenovi.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+export const handler = Sentry.AWSLambda.wrapHandler(async (event: APIGatewayEvent) => {
+  if (!event.body) {
+    console.log("Event has no body - will not be forwarded to the API");
+    return defaultResponse();
+  }
+
+  const authHeader: string = (await getSecret(tenoviAuthHeader)) as string;
+  if (!authHeader) {
+    throw new Error(`Config error - TENOVI_AUTH_HEADER doesn't exist`);
```
Comment (leite08) [inline packages/lambdas/src/tenovi.ts:32]: This error message leads me to think the env var is missing, which is not the case here. Maybe indicate that the secret was not found or something?

**Review by leite08 — COMMENTED** [review]
Mostly LGTM, just a few comments

File: packages/api/src/mappings/tenovi/body.ts (hunk 1 above)
Comment (RamilGaripov) [reply packages/api/src/mappings/tenovi/body.ts]: Yea def good practice to do that!

**Review by leite08 — APPROVED** [review]

**Pull request merged**